build-tools/packages/build-tools/src/common/npmPackage.ts

```diff
@@ -477,24 +476,48 @@ export class Packages {
  * Reads the contents of package.json, applies a transform function to it, then writes the results back to the source
  * file.
  *
- * @param packageDir - The path to the directory containing package.json.
+ * @param packagePath - A path to a package.json file or a folder containing one. If the path is a directory, the
+ * package.json from that directory will be used.
  * @param packageTransformer - A function that will be executed on the package.json contents before writing it
  * back to the file.
  *
  * @remarks
  *
  * The package.json is always sorted using sort-package-json.
+ *
+ * @internal
  */
 export function updatePackageJsonFile(
-	packageDir: string,
+	packagePath: string,
```

Comment:
I made this change primarily for testing. I want to test this function against a test file, but if that file is named package.json, pnpm thinks they're packages in the workspace, and I can't change the workspace definition to exclude them because fluid-build doesn't support the full range of globs that pnpm does. With this change I can name the files `_package.json` which keeps everyone happy.